extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -37,6 +38,83 @@ class AmpStickyAd extends AMP.BaseElement {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+
+    const children = this.getRealChildren();
+    user.assert((children.length == 1 && children[0].tagName == 'AMP-AD'),
+        'amp-sticky-ad must have a single amp-ad child');
+    /** @const @private {!Element} */
+    this.ad_ = children[0];
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = this.getViewport();
+
+    /**
+     * On viewport scroll, check requirements for amp-stick-ad to display.
+     * @const @private {!Unlisten}
+     */
+    this.scrollUnlisten_ =
+        this.viewport_.onScroll(() => this.displayAfterScroll_());
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
+    if (!this.isExperimentOn_) {
+      dev.warn(TAG, `TAG ${TAG} disabled`);
+      return Promise.resolve();
+    }
+    return Promise.resolve();
+  }
+
+  /** @override */
+  detachedCallback() {
+    this.removeOnScrollListener();
+  }
+
+  /**
+   * The function that remove listener to viewport onScroll event.
+   * @private
+   */
+  removeOnScrollListener() {
+    this.scrollUnlisten_();
+    this.scrollUnlisten_ = null;
+  }
+
+  /**
+   * The listener function that listen on onScroll event and
+   * show sticky ad when user scroll at least one viewport and
+   * there is at least one more viewport available.
+   * @private
+   */
+   // TODO(zhouyx): not really a member variable, need to move
```

Comment:
What's not a member variable? Why TODO? Also, jsdoc and method declarations cannot be interrupted by comments.